arch/arm64/core/fatal.c

```diff
@@ -350,7 +351,10 @@ FUNC_NORETURN void arch_system_halt(unsigned int reason)
 	ARG_UNUSED(reason);
 
 	(void)arch_irq_lock();
-	(void)pm_system_off();
+
+#ifdef CONFIG_SHUTDOWN
+	sys_shutdown();
+#endif /* CONFIG_SHUTDOWN */
```

Comment:
btw, should arch_system_halt be calling shutdown? IIRC halt means to stop the system, but not cut the power.